klp_test_support_mod.h

```diff
@@ -130,8 +131,12 @@ int PATCHED_SYM(do_sleep)(unsigned long secs, int task_state)
 	hrtimer_init_on_stack(&t.timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
 	hrtimer_set_expires_range_ns(&t.timer, ktime_set(secs, 0), 0);
 	hrtimer_init_sleeper(&t, current);
+#else
+#ifdef USE_NEW_HRTIMER_SETUP_API
+	hrtimer_setup_sleeper_on_stack(&t, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
 #else
 	hrtimer_init_sleeper_on_stack(&t, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+#endif
 	hrtimer_set_expires_range_ns(&t.timer, ktime_set(secs, 0), 0);
```

Comment:
I would prefer to separate it completely.

That is to have

#if OLD
  hrtimer_init_on_stack()
  hrtimer_set_expires_range_ns()
  hrtimer_init_sleeper
#elif NEW
  hrtimer_setup_sleeper_on_stack()
  hrtimer_set_expires_range_ns()
#else 
  hrtimer_init_sleeper_on_stack()
  hrtimer_set_expires_range_ns()
#endif

but see the comments above.

Reply:
sure